### PR TITLE
Unit and integration tests for procedure

### PR DESCRIPTION
**Pull request by sfc-gh-jsikorski**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
* Added integration and unit tests for snowpark procedure
*  Separated and reformatted integration tests steps for both snowpark function and procedure,
* Minor fixes for test_utils.py

### CODE REVIEW

File: tests/test_procedure.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    DIR_INITIAL_CONTENTS = {
+        "requirements.txt",
+        "local_connection.py",
+        ".gitignore",
+        "app.py",
+        "config.toml",
+    }
+
+    def test_procedure_init(self, tmp_dir_for_procedure_tests):
+        procedure.procedure_init()
```
Comment (sfc-gh-turbaszek) [inline tests/test_procedure.py:25]: I think in other unit tests we use `runner` to invoke the command instead calling the method directly. I think we should decided witch approach we want to go.

File: tests_integration/test_procedure.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+)
+from tests_integration.testing_utils.sql_utils import sql_test_helper
+from tests_integration.testing_utils.working_directory_utils import (
+    temporary_working_directory,
+)
+
+
+@pytest.mark.integration
+def test_snowpark_procedure_flow(_test_steps):
+    _test_steps.assert_that_no_entities_are_in_snowflake()
```
Comment (sfc-gh-turbaszek) [inline tests_integration/test_procedure.py:20]: what entities? There will be entities like accounts, roles. Should we be more explicite?

File: tests_integration/test_procedure.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    _test_steps.assert_that_no_files_are_staged_in_snowflake()
```
Comment (sfc-gh-turbaszek) [inline tests_integration/test_procedure.py:21]: Staged anywhere or on particular stage?

Comment (sfc-gh-jsikorski) [reply tests_integration/test_procedure.py]: I changed "snowflake" to "test_db" - hope this will be clearer

File: tests/test_procedure.py (hunk 1 above)
Comment (sfc-gh-jsikorski) [reply tests/test_procedure.py]: Done

File: tests/test_procedure.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    TEMP_DIRECTORY_NAME = "test_procedure_tmp"
+    DIR_INITIAL_CONTENTS = {
+        "requirements.txt",
+        "local_connection.py",
+        ".gitignore",
+        "app.py",
+        "config.toml",
+    }
+
+    def test_procedure_init(self, tmp_dir_for_procedure_tests, runner):
```
Comment (sfc-gh-astus) [inline tests/test_procedure.py:24]: This method is used anywhere?

File: tests/test_procedure.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+from zipfile import ZipFile
+
+import pytest
+
+import snowcli.utils
+from snowcli.cli.snowpark import procedure
+from snowcli.utils import SplitRequirements
+
+
+class TestProcedure:
```
Comment (sfc-gh-astus) [inline tests/test_procedure.py:14]: I'm not sure about this class, usually we write tests without class

File: tests_integration/test_procedure.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply tests_integration/test_procedure.py:20]: I would suggest `_assert_no_procedures_in_test_schema`

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: tests_integration/test_procedure.py (hunk 2 above)
Comment (sfc-gh-jsikorski) [reply tests_integration/test_procedure.py:20]: What about functions then? This test step is used for both

**Pull request merged**